interview-questions.md

```diff
@@ -7,52 +7,52 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What are props in React?
 
-  Your answer:
+  Your answer: Props, also known as properties, are passed down from the parent page to the child component pages. They all for data to be shared across all components.
 
-  Researched answer:
+  Researched answer: Props are arguments that are passed into React components. Using the same syntax as HTML, the component recieves the data as an argument.
 
 
 
 2. What is a DOM event?
 
-  Your answer:
+  Your answer: An event that accures on the actual page the user is on. It can be defined as a user event. Clicking on options and filling in data are examples of this.
 
-  Researched answer:
+  Researched answer: A signal that an action has taken place. It is trigged by the user or by hte browser. 
 
 
 
 3. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: A programming language that's based around objects. It's structure strictly uses objects, favoring organization of data and methods. Functional programming is more open and allows for user created functions that can be stored in custom named variables. This also allows for user created classes and objects.
 
-  Researched answer:
+  Researched answer: Where as object oriented programming gives importance to data rather than procedures, functional programming does just the opposite. Object oriented programming is based around classes and objects. Functional programming favors the use of functions. Functions that are created for a specific task.
 
 
 
 4. What is the difference between a Float and an Integer in Ruby?
 
-  Your answer:
+  Your answer: An integer is a number value without decimals. A float is a number that uses a decimal syntax and is used for more precise computing. 
 
-  Researched answer:
+  Researched answer: Integers are whole numbers that can be both positive and negative while a float number is a real number that uses decimal points. 
 
 
 
 5. Ruby has an implicit return. What does that mean?
 
-  Your answer:
+  Your answer: A function that gives an automatic return
 
-  Researched answer:
+  Researched answer: A function that runs without the use of curly braces. This syntax gives you an implied return
```

Comment:
The implicit return is more about the keyword return (or lack there of) than the curly braces.